pennylane/gradients/jvp.py

```diff
@@ -0,0 +1,287 @@
+        # TODO remove the pragma in PR #3170
+    except (AttributeError, TypeError):  # pragma: no cover
+        pass
+
+    gradient_tapes, fn = gradient_fn(tape, **gradient_kwargs)
+
+    def processing_fn(results):
+        # postprocess results to compute the Jacobian
+        jac = fn(results)
+        if multi_m:
+            return compute_jvp_multi(tangent, jac)
+        return compute_jvp_single(tangent, jac)
+
+    return gradient_tapes, processing_fn
+
+
+def batch_jvp(tapes, tangents, gradient_fn, reduction="append", gradient_kwargs=None):
+    r"""Generate the gradient tapes and processing function required to compute
+    the Jacobian vector products of a batch of tapes.
+    Args:
+        tapes (Sequence[.QuantumTape]): sequence of quantum tapes to differentiate
+        tangentss (Sequence[tensor_like]): Sequence of gradient-output vectors ``dy``. Must be the
+            same length as ``tapes``. Each ``dy`` tensor should have shape
+            matching the output shape of the corresponding tape.
+        gradient_fn (callable): the gradient transform to use to differentiate
+            the tapes
+        reduction (str): Determines how the Jacobian-vector products are returned.
+            If ``append``, then the output of the function will be of the form
+            ``List[tensor_like]``, with each element corresponding to the VJP of each
+            input tape. If ``extend``, then the output VJPs will be concatenated.
+        gradient_kwargs (dict): dictionary of keyword arguments to pass when
+            determining the gradients of tapes
+    Returns:
+        List[tensor_like or None]: list of Jacobian vector products. ``None`` elements corresponds
+        to tapes with no trainable parameters.
+    **Example**
+    # TODO: add examples
+    """
+    gradient_kwargs = gradient_kwargs or {}
+    reshape_info = []
+    gradient_tapes = []
+    processing_fns = []
+
+    # Loop through the tapes and dys vector
+    for tape, tangent in zip(tapes, tangents):
+        g_tapes, fn = jvp(tape, tangent, gradient_fn, gradient_kwargs)
+
+        reshape_info.append(len(g_tapes))
+        processing_fns.append(fn)
+        gradient_tapes.extend(g_tapes)
+
+    def processing_fn(results):
+        jvps = []
+        start = 0
+
+        for t_idx in range(len(tapes)):
+            # extract the correct results from the flat list
+            res_len = reshape_info[t_idx]
+            res_t = results[start : start + res_len]
+            start += res_len
+
+            # postprocess results to compute the VJP
+            jvp_ = processing_fns[t_idx](res_t)
+
+            if jvp_ is None:
+                if reduction == "append":
+                    jvps.append(None)
+                continue
+
+            if isinstance(reduction, str):
+                if reduction == "extend":
+                    if not isinstance(jvp_, tuple) and jvp_.shape == ():
+                        jvp_ = math.reshape(jvp_, (1,))
```

Comment:
Instead of having this special case here, I think it would be better to instead pass the appropriate `reduction` function from the caller of `batch_jvp`. So it would look something like
```python
vjps = batch_vjp(
    ...,
    reduction=lambda vjps, x: vjps.extend(qml.math.reshape(x, (1,)) if ... else x),
    ...
)
```

Reply:
Thanks for the idea, I am updating it!